examples/gtest/CMakeLists.txt

```diff
@@ -2,6 +2,10 @@ cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
 
 project(CPMExampleGtest)
 
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS NO)
```

Comment:
Are these global `set`s actually necessary?

Reply:
without this, it does not compile on my host

Reply:
It does work on my system and CI, so there is something strange going on here. What version of MacOS, CMake, and clang are you using?

Reply:
cmake version 3.19.4

Darwin Claus-iMac.fritz.box 20.2.0 Darwin Kernel Version 20.2.0: Wed Dec  2 20:39:59 PST 2020; root:xnu-7195.60.75~1/RELEASE_X86_64 x86_64

I think there are problems with different version of gtest header installed on my host.
I will check this